src/Infrastructure/Repository/TaskYamlRepository.php

```diff
@@ -406,26 +406,55 @@ protected function buildTask(array $taskData, array $taskListData, array $tags =
      * @param array<string, \SprykerSdk\SdkContracts\Entity\TaskInterface> $tasks
      * @param array $tags
      *
-     * @return \SprykerSdk\SdkContracts\Entity\TaskInterface
+     * @return \SprykerSdk\Sdk\Core\Domain\Entity\Task
      */
-    protected function buildTaskSet(array $taskData, array $taskListData, array $tasks, array $tags = []): TaskInterface
+    protected function buildTaskSet(array $taskData, array $taskListData, array $tasks, array $tags = []): Task
     {
         $task = $this->buildTask($taskData, $taskListData, $tags);
 
         if (!isset($taskData['tasks'])) {
             return $task;
         }
 
-        foreach ($taskData['tasks'] as $taggedTaskData) {
-            $taggedTask = $tasks[$taggedTaskData['id']] ?? null;
+        $taskSetCommands = [];
+        $taskSetPlaceholders = [];
 
-            if (!$taggedTask instanceof TaggedTaskInterface) {
+        foreach ($taskData['tasks'] as $subTaskData) {
+            $subTask = $tasks[$subTaskData['id']] ?? null;
+
+            if ($subTask === null) {
                 continue;
             }
 
-            $taggedTask->setTags($taggedTaskData['tags'] ?? []);
+            $subTaskCommands = $this->addStageToSubTasks($subTask, $subTask->getCommands());
+
+            $taskSetCommands = array_merge($subTaskCommands, $taskSetCommands);
```

Comment:
move array_merge from foreach. you can use it like
$taskSetCommands[] = subTaskCommands; in foreach and then

$taskSetCommands = array_merge(...$taskSetCommands)